File: trunk/components/src/main/java/org/wtk/feedback/handler/AlertFeedbackHandler.java
package org.wtk.feedback.handler;

import org.wtk.feedback.dispatcher.BaseJsFeedbackHandler;

/**
 * @author Yoav Aharoni
 */
public class AlertFeedbackHandler extends BaseJsFeedbackHandler {
	private static final AlertFeedbackHandler INSTANCE = new AlertFeedbackHandler();

	private String format;

	private AlertFeedbackHandler() {
		this("level + ' - ' + message");
	}

	private AlertFeedbackHandler(String format) {
		super(AlertFeedbackHandler.class, "AlertHandler");
		addJavaScript().setTemplateModel(this);
		this.format = format;
	}

	public String getFormat() {
		return format;
	}

	public static AlertFeedbackHandler get() {
		return INSTANCE;
	}

	public static AlertFeedbackHandler get(String format) {
		return new AlertFeedbackHandler(format);
	}
}

File: trunk/components/src/main/java/org/wtk/feedback/handler/AlertFeedbackHandler.js
wtk.feedback.AlertHandler = wtk.clazz(wtk.feedback.Handler, {
	priority: 0,

	initialize: function(format) {
		this.$super.initialize.apply(this, arguments);
		this.format = format;
	},

	onFeedback: function(messages) {
		var text = '';

		wtk.each(messages, function(msg) {
			text += this.format(msg.message, msg.level, msg.reporter) + '\n';
		}, this);

		if (text) {
			alert(text);
		}
	}
});